code/game/objects/items/melee/misc.dm

```diff
@@ -1079,3 +903,92 @@
+	var/non_harm_force = 4
+	/// Stamina damage dealt
+	var/stamina_force = 25
+
+/obj/item/melee/tonfa/attack(mob/living/target, mob/living/user)
+	var/target_zone = user.get_combat_bodyzone(target)
+	var/armour_level = target.getarmor(target_zone, STAMINA, penetration = armour_penetration - 15)
+
+	add_fingerprint(user)
+	if((HAS_TRAIT(user, TRAIT_CLUMSY)) && prob(50))
+		to_chat(user, "<span class ='danger'>You hit yourself over the head.</span>")
+		user.adjustStaminaLoss(stamina_force)
+
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			H.apply_damage(2*force, BRUTE, BODY_ZONE_HEAD)
+		else
+			user.take_bodypart_damage(2*force)
+		return
+	if(!isliving(target))
+		return ..()
+	if(iscyborg(target))
+		if (user.a_intent != INTENT_HARM)
+			playsound(get_turf(src), hitsound, 75, 1, -1)
+			user.do_attack_animation(target) // The attacker cuddles the Cyborg, awww. No damage here.
+			return
+	if (user.a_intent != INTENT_HARM)
+		force = non_harm_force
+	else
+		force = initial(force)
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if (H.check_shields(src, 0, "[user]'s [name]", MELEE_ATTACK))
+			return
+		if(check_martial_counter(H, user))
+			return
+
+		var/mob/living/carbon/human/T = target
+
+		target.visible_message("[user] strikes [target] in the [parse_zone(target_zone)].", "You strike [target] in the [parse_zone(target_zone)].")
+		log_combat(user, target, "attacked", src)
+
+		// If the target has a lot of stamina loss, knock them down
+		if ((user.is_zone_selected(BODY_ZONE_L_LEG) || user.is_zone_selected(BODY_ZONE_R_LEG)) && target.getStaminaLoss() > 22)
+			var/effectiveness = CLAMP01((target.getStaminaLoss() - 22) / 50)
+			log_combat(user, target, "knocked-down", src, "(additional effect)")
+			// Move the target back upon knockdown, to give them some time to recover
+			var/shove_dir = get_dir(user.loc, target.loc)
+			var/turf/target_shove_turf = get_step(target.loc, shove_dir)
+			var/mob/living/carbon/human/target_collateral_human = locate(/mob/living/carbon) in target_shove_turf.contents
+			if (target_collateral_human && target_shove_turf != get_turf(user))
+				target.Knockdown(max(0.5 SECONDS, effectiveness * 4 SECONDS * (100-armour_level)/100))
+				target_collateral_human.Knockdown(0.5 SECONDS)
+			else
+				target.Knockdown(effectiveness * 4 SECONDS * (100-armour_level)/100)
+			target.Move(target_shove_turf, shove_dir)
+		if (user.is_zone_selected(BODY_ZONE_L_LEG) || user.is_zone_selected(BODY_ZONE_R_LEG) || user.is_zone_selected(BODY_ZONE_L_ARM) || user.is_zone_selected(BODY_ZONE_R_ARM))
+			// 4-5 hits on an unarmoured target
+			target.apply_damage(stamina_force*0.6, STAMINA, target_zone, armour_level)
+		else
+			// 4-5 hits on an unarmoured target
+			target.apply_damage(stamina_force, STAMINA, target_zone, armour_level)
+
+		add_fingerprint(user)
+
+		if(!iscarbon(user))
+			target.LAssailant = null
+		else
+			target.LAssailant = WEAKREF(user)
```

Comment:
I think It looks it should be called when it's early returned from other lines.

Reply:
most other early returns aren't actually performing an attack, this exists for slime shitcode that I am going to make another PR removing

Reply:
I see the PR.